modules/analysis-common/src/main/java/org/elasticsearch/analysis/common/SynonymTokenFilterFactory.java

```diff
@@ -62,8 +76,10 @@ public class SynonymTokenFilterFactory extends AbstractTokenFilterFactory {
                     + "Instead, insert a lowercase filter in the filter chain before the synonym_graph filter."
             );
         }
+
+        this.synonymsSource = parseSettingsForSynonymsSource(settings);
         this.expand = settings.getAsBoolean("expand", true);
-        this.lenient = settings.getAsBoolean("lenient", false);
+        this.lenient = settings.getAsBoolean("lenient", this.synonymsSource == SynonymsSource.INDEX);
```

Comment:
This changes the default `lenient` value to `true` when using `synonyms_set`, but the user can still get into trouble if they explicitly set `lenient` to `false`. Do we want to throw an error on that?

Reply:
I'd say to keep the setting as true by default in all types - that would be better from my PoV to avoid having users to understand that this setting depends on the type. And it's the safest thing to do for file based as well.

Reply:
I like the idea of defaulting to `true` but I also think we could document this - that if `lenient` is set to `false` you may run into this behavior.

Reply:
I'm going to set the default to true for all cases, barring any new feedback against this decision